lib/extract_field/FieldExtractorManager.js

```diff
@@ -0,0 +1,40 @@
+const FieldExtractor = require("perimeterx-node-core/lib/extract_field/FieldExtractor");
+const ExtractionField = require("perimeterx-node-core/lib/models/ExtractionField");
+
+const USERNAME_FIELD = "user";
+const PASSWORD_FIELD = "pass";
+
+class FieldExtractorManager { 
+    constructor(extractObjects) {
+        this.extractorMap = this._createExtractorsMap(extractObjects);
+    }
+
+    ExtractFields(request) {
+        const key = this._getMapKey(request.path, request.method);
+        const extractor = this.extractorMap[key];
+        try {
+            return extractor ? extractor.ExtractFields(request) : {};
+        } catch (e) {
+            console.error(e);
+            return {};
+        }
+    }
+
+    _getMapKey(requestPath, requestMethod) {
+        return `${requestPath}:${requestMethod}`;
+    }
+
+    _createExtractorsMap(extractObjects) {
+        const map = {};
+        for (const extractFields of extractObjects) {
+            const key = this._getMapKey(extractFields.path, extractFields.method.toUpperCase());
+            map[key] = new FieldExtractor(extractFields.sentThrough, extractFields.contentType, extractFields.encoding, [
+                new ExtractionField(extractFields.userField, USERNAME_FIELD), 
```

Comment:
why do you need this nested structure, what is the benefit of adding _ExtractionField_ object, wasn't it enough to just use _FieldExtractor_ object and add userField and outputfield (ExtractionField attributes) to FieldExtractor object?

Reply:
I know what will be your answer lol, that you must connect the path to the field to the output name in the activity and you wrapped it in an object.

Reply:
Exactly :)